alica_engine/src/engine/RuleBook.cpp

```diff
@@ -157,6 +157,13 @@ PlanChange RuleBook::dynamicAllocationRule(RunningPlan& r)
     if (!r.getCycleManagement().mayDoUtilityCheck()) {
         return PlanChange::NoChange;
     }
+
+    const Plan* p = static_cast<const Plan*>(r.getActivePlan());
+
+    if (p->getUtilityThreshold() < 0) {
```

Comment:
`0`  -> `0.` due function returns double
we don't want to exclude case with getUtilityThreshold() == 0.?

Reply:
for 0, we should not be returning here. Instead 0 means if we find a better assignment we should use it even if the assignment's utility is only slightly better.